src/ostree/ot-builtin-summary.c

```diff
@@ -40,6 +40,63 @@ static GOptionEntry options[] = {
   { NULL }
 };
 
+/* Take arguments of the form [key=value] or [key, value] out of @argv and put
+ * them into an a{sv} variant. The value arguments must be parsable using
+ * g_variant_parse(). */
```

Comment:
> is there a reason not to just require the user to pass the whole `a{sv}` that way?

I imagine that would get a bit faffy with keys which are optional (for example, the caller always wants to pass in `key1` and `key2`, but `key3` and `key4` are conditional on different things).

> We have some prior art here for `ostree commit --add-metadata-string`; do we expect non-string keys here?

Yeah, I think integers (timestamps) will be fairly common. `ostree.summary.expiry`, for example.